dwa_local_planner/src/dwa_planner_ros.cpp

```diff
@@ -439,5 +449,31 @@ namespace dwa_local_planner {
 
   }
 
+  bool DWAPlannerROS::backwardDistanceExceeded(const geometry_msgs::PoseStamped& robot_pose, double linear_speed) {
+    auto distance = [](const geometry_msgs::PoseStamped& pose1, const geometry_msgs::PoseStamped& pose2) {
+      return hypot(pose2.pose.position.x - pose1.pose.position.x, pose2.pose.position.y - pose1.pose.position.y);
+    };
 
+    // TODO: I can have a queue of speeds and only reset when all are >= 0
+    if ( ! bw_motion_start_pose_) {
+      if (linear_speed < 0.0) {  // TODO: should I use planner_util_.getCurrentLimits().min_vel_trans instead of 0?
```

Comment:
should I use planner_util_.getCurrentLimits().min_vel_trans instead of 0?

Reply:
no; the robot can move at a backwards speed slower than `min_vel_trans` as long as the rotational velocity is above `min_vel_trans`